plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavaxToolsCompiler.java

```diff
@@ -57,7 +58,7 @@ protected JavaCompiler newJavaCompiler()
 		return ToolProvider.getSystemJavaCompiler();
 	}
 
-    private List<JavaCompiler> JAVA_COMPILERS = new CopyOnWriteArrayList<>();
+    private final List<JavaCompiler> JAVA_COMPILERS = new CopyOnWriteArrayList<>();
```

Comment:
a class field used in a synchronized block, this is the place where `final` is important and it was not there